### PR TITLE
Resolves #192 - Use green color for successful build card theme.

### LINKED ISSUE #192: Advice request: how to implement option to use green color for SUCCESS builds
Hi @damianszczepanik (and anyone who wants to paticipate)
I want to implement ability to use green color for notification from successful build and I ask for your opinion how.

Green is more "natural" for Jenkins-unaware folks, e.g. analysts, testers, sales people - so notifications from this plugin would be more readable also for "non-build" pipelines.

**Possible implementations**
1. Simplest - add checkbox to "Use green color for Success result"
2. Still simple - add checkbox to "Override color for Success result", which shows Input to specify the color
3. Advanced - implement "Jenkins extenstions" which modify the behavior of the plugin - like with Git Plugin you can select additional behavior like "Use shallow clone"

Thanks in advance any feedback.

### PR DESCRIPTION
**Pull request by BradPatras**
Didn't see any tests that need updated.  
I would have liked to consolidate the ```getHTMLBaseColor()``` method into a Util class so that the string creation logic can be shared but that change would have been out of the scope of this repository.

### CODE REVIEW

**Comment by BradPatras** [comment]
Need to update the request JSON theme color to ```#00FF00``` to pass the successful build tests.

File: src/main/java/jenkins/plugins/office365connector/CardBuilder.java [hunk 1]
```diff
@@ -87,12 +88,20 @@ public Card createCompletedCard(List<FactDefinition> factDefinitions) {
...
 
         Card card = new Card(summary, section);
-        card.setThemeColor(lastResult.color.getHtmlBaseColor());
+        card.setThemeColor(getCardThemeColor(lastResult));
         card.setPotentialAction(potentialActionBuilder.buildActionable());
 
         return card;
     }
 
+    private String getCardThemeColor(Result result) {
```
Comment (damianszczepanik) [inline src/main/java/jenkins/plugins/office365connector/CardBuilder.java]: add unit test for this method

**Comment by BradPatras** [comment]
Removed redundant null check which I believe is why those tests failed

File: src/main/java/jenkins/plugins/office365connector/CardBuilder.java [hunk 2]
```diff
@@ -94,8 +94,8 @@ public Card createCompletedCard(List<FactDefinition> factDefinitions) {
         return card;
     }
 
-    private String getCardThemeColor(Result result) {
-        if (result == Result.SUCCESS) {
+    private static String getCardThemeColor(Result result) {
+        if (result != null && result == Result.SUCCESS) {
```
Comment (damianszczepanik) [inline src/main/java/jenkins/plugins/office365connector/CardBuilder.java]: what is the reason of having this additional comparison?

File: src/test/java/jenkins/plugins/office365connector/CardBuilderTest.java [hunk 3]
```diff
@@ -505,4 +506,54 @@ public void getEscapedDisplayName_OnNameWithSpecialCharacters_EscapesSpecialChar
         // then
         assertThat(displayName).isEqualTo("this\\_is\\_my\\-very\\#special \\*job\\*");
     }
+
+    @Test
+    public void getCardThemeColor_OnSuccessResult_ReturnsGreen() {
+        Result successResult = Result.SUCCESS;
+        String themeColor = Deencapsulation.invoke(CardBuilder.class, "getCardThemeColor", successResult);
+        
+        String greenColorString = String.format("#%06X", Color.GREEN.getRGB() & 0xFFFFFF);
```
Comment (damianszczepanik) [inline src/test/java/jenkins/plugins/office365connector/CardBuilderTest.java]: I think this is to complicated to get green color :)

File: src/test/java/jenkins/plugins/office365connector/CardBuilderTest.java [hunk 4]
```diff
@@ -505,4 +506,54 @@ public void getEscapedDisplayName_OnNameWithSpecialCharacters_EscapesSpecialChar
         // then
         assertThat(displayName).isEqualTo("this\\_is\\_my\\-very\\#special \\*job\\*");
     }
+
+    @Test
+    public void getCardThemeColor_OnSuccessResult_ReturnsGreen() {
+        Result successResult = Result.SUCCESS;
```
Comment (damianszczepanik) [inline src/test/java/jenkins/plugins/office365connector/CardBuilderTest.java]: add given/when/then to clarify what are you testing

File: src/main/java/jenkins/plugins/office365connector/CardBuilder.java (hunk 2 above)
Comment (BradPatras) [reply src/main/java/jenkins/plugins/office365connector/CardBuilder.java]: No reason, my mistake for keeping this in.

File: src/test/java/jenkins/plugins/office365connector/CardBuilderTest.java (hunk 3 above)
Comment (BradPatras) [reply src/test/java/jenkins/plugins/office365connector/CardBuilderTest.java]: I agree, I will use the actual string instead of that method.

**Comment by damianszczepanik** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/damianszczepanik/office-365-connector-plugin/pullRequest?prid=5132103&bid=16864531)

File: src/test/java/jenkins/plugins/office365connector/CardBuilderTest.java (hunk 4 above)
Comment (BradPatras) [reply src/test/java/jenkins/plugins/office365connector/CardBuilderTest.java]: I was not previously familiar with given/when/then, but I believe I added them properly

**Pull request merged**